src/mlpack/core/data/dataset_mapper_impl.hpp

```diff
@@ -230,6 +230,31 @@ inline size_t DatasetMapper<PolicyType, InputType>::Dimensionality() const
   return types.size();
 }
 
+// Remove a dimension
+template<typename PolicyType, typename InputType>
+inline void DatasetMapper<PolicyType, InputType>::RemoveDimension(
+    const size_t dimension)
+{
+  if (dimension >= types.size())
+  {
+    std::ostringstream oss;
+    oss << "requested to remove dimension " << dimension << ", but dataset only "
+        << "has " << types.size() << " dimensions";
```

Comment:
```suggestion
    oss << "DatasetMapper::RemoveDimension(): requested to remove dimension "
        << dimension << ", but dataset only has " << types.size()
        << " dimensions";
```
This can be helpful to be clear about where the exception is being thrown from.